### PR TITLE
Integrate Yandex.Translate engine

### PR DESCRIPTION
**Pull request by v4ernenko**
https://translate.yandex.com/developers

### CODE REVIEW

**Comment by Ostico** [comment]
Hi @v4ernenko ,

thank you for the pull request, it will take a little longer than usual because we have to test it and now we are near to the Christmas holidays. 

Most likely we will be able to publish it around January 15th. I hope this is OK with you.

**Comment by v4ernenko** [comment]
OK, thank you for the update!

**Comment by Lilimopps** [comment]
Thank you! Irina 21.12.2016, 17:46, "Vadim Chernenko" <notifications@github.com>:OK, thank you for the update!—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.   Regards,Irina Rybnikova

File: lib/Utils/Engines/YandexTranslate.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            } else {
+                $decoded = array(
+                    'error' => array(
+                        'code' => $decoded[ "code" ],
+                        'message' => $decoded[ "message" ]
+                    )
+                );
+            }
+        } else {
+            $decoded = $rawValue; // already decoded in case of error
```
Comment (v4ernenko) [inline lib/Utils/Engines/YandexTranslate.php]: Fixed! Thank you!

**Comment by Ostico** [comment]
Well done :+1:

**Comment by Lilimopps** [comment]
Please find some answers below.

> First of all, can you send me a test key to access your engine?

You can get test key here: https://tech.yandex.com/keys/get/?service=trnsl

> Moreover, there are some kind of limitations in throughput, on the connection limits or on the concurrency?

We recommend no more than 10 active connections at any one time.

> What are the supported languages?

You can find the list of supported languages here: https://tech.yandex.com/translate/doc/dg/concepts/api-overview-docpage/
or use getLangs method: https://tech.yandex.com/translate/doc/dg/reference/getLangs-docpage/

We haven't yet implemented a language support check now. But we can do it, if needed.

**Comment by Lilimopps** [comment]
@Ostico What are the next steps? Can we go on public now?

**Pull request merged**